packages/console/src/notebook/ScriptEditor.tsx

```diff
@@ -1,19 +1,51 @@
 /**
  * Script editor for large blocks of code
  */
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { Component, ReactElement, RefObject } from 'react';
 import { LoadingOverlay, ShortcutRegistry } from '@deephaven/components';
 import Log from '@deephaven/log';
+import { IdeSession } from '@deephaven/jsapi-shim';
+import { assertNotNull } from '@deephaven/utils';
+import { editor, IDisposable } from 'monaco-editor';
 import Editor from './Editor';
 import { MonacoCompletionProvider, MonacoUtils } from '../monaco';
 import './ScriptEditor.scss';
 import SHORTCUTS from '../ConsoleShortcuts';
 
 const log = Log.module('ScriptEditor');
 
-class ScriptEditor extends Component {
-  constructor(props) {
+interface ScriptEditorProps {
+  error?: { message?: string };
+  isLoading: boolean;
+  isLoaded: boolean;
+  focusOnMount?: boolean;
+  onChange: (e: editor.IModelContentChangedEvent) => void;
+  onRunCommand: (command: string | null) => void;
```

Comment:
Don't allow `null` here
```suggestion
  onRunCommand: (command: string) => void;
```

Reply:
The actual function passed in does accept values of null though